android-benchmark/benchmark/src/androidTest/kotlin/com/uber/rxdogtag/androidbenchmark/RxDogTagAndroidPerf.kt

```diff
@@ -39,214 +43,62 @@ import java.util.concurrent.CountDownLatch
  * Basic command line usage is just to run `./gradlew :android-benchmark:benchmark:connectedCheck`
  */
 @LargeTest
-class RxDogTagAndroidPerf {
+@RunWith(Parameterized::class)
+class RxDogTagAndroidPerf(private val enabled: Boolean, private val times: Int) {
+
+  companion object {
+    @JvmStatic
+    @Parameters(name = "enabled={0},times={1}")
+    fun data() = listOf(
+        arrayOf(true, 0),
```

Comment:
Correct me if I'm wrong, but wouldn't we be generating lot of tests that we don't care about in this parameterized model? 

For example, I'll have a boolean for whether guardedDelegateCall is run. 

That will generate a redundant set permutation of RxDogTag being disabled and guardedDelegate being enabled. 

This will only increase as we add more parameters.

Reply:
When that time comes, it's easy to add an annotation + rule that ignores tests for certain configurations. Example: https://github.com/square/kotlinpoet/blob/master/kotlinpoet/src/test/java/com/squareup/kotlinpoet/KmSpecsTest.kt#L84

Reply:
Cool, that's neat.